### PR TITLE
TXT: Fix warnings in tests

### PR DESCRIPTION
**Pull request by hong2223**
Ready for the review!

### CODE REVIEW

File: trackintel/model/util.py [hunk 1]
```diff
@@ -31,7 +33,10 @@ def get_speed_positionfixes(positionfixes):
     g = pfs.geometry
     # get distance and time difference
     if is_planar_crs:
-        dist = g.distance(g.shift(1)).to_numpy()
+        with warnings.catch_warnings():
+            warnings.filterwarnings("ignore", message="CRS not set for some of the concatenation inputs.*")
```
Comment (NinaWie) [inline trackintel/model/util.py:37]: Why are the warnings filtered here? Isn't it a problem if the crs is not set for the positionfixes?

File: trackintel/preprocessing/staypoints.py [hunk 2]
```diff
@@ -270,7 +270,10 @@ def merge_staypoints(staypoints, triplegs, max_time_gap="10min", agg={}):
 
     # a joined dataframe sp_tpls is constructed to add the columns 'type' and 'next_type' to the 'sp_merge' table
     # concat and sort by time
-    sp_tpls = pd.concat([sp_merge, tpls_merge]).sort_values(by=["user_id", "started_at"])
+    with warnings.catch_warnings():
+        warnings.filterwarnings("ignore", message="CRS not set for some of the concatenation inputs.*")
+
+        sp_tpls = pd.concat([sp_merge, tpls_merge]).sort_values(by=["user_id", "started_at"])
```
Comment (NinaWie) [inline trackintel/preprocessing/staypoints.py:276]: This is related to the issue you opened, right? I think it would be better to have a separate case for the empty tpls case, as you suggested :)

File: trackintel/preprocessing/triplegs.py [hunk 3]
```diff
@@ -132,7 +132,9 @@ def _seperate_ids(row):
     trips.drop(columns=["type", "sp_tpls_id"], inplace=True)  # make space so no overlap with activity "sp_tpls_id"
     # Inserting `gaps` and `user_change` into the dataframe creates buffers that catch shifted
     # "staypoint_id" and "trip_id" from corrupting staypoints/trips.
-    trips_with_act = pd.concat((trips, sp_tpls_only_act, gaps, user_change), axis=0, ignore_index=True)
+    with warnings.catch_warnings():
+        warnings.filterwarnings("ignore", message="CRS not set for some of the concatenation inputs.*")
+        trips_with_act = pd.concat((trips, sp_tpls_only_act, gaps, user_change), axis=0, ignore_index=True)
```
Comment (NinaWie) [inline trackintel/preprocessing/triplegs.py:137]: Is this also related to empty dataframes?

File: trackintel/model/util.py (hunk 1 above)
Comment (NinaWie) [reply trackintel/model/util.py:37]: Ah, is this problem due to the g.shift? After it's shifted, there is no crs defined anymore?

**Review by NinaWie — COMMENTED** [review]
Awesome that so many of the warnings are gone! 
I had a few questions but looks good in general :)

File: trackintel/preprocessing/triplegs.py (hunk 3 above)
Comment (hong2223) [reply trackintel/preprocessing/triplegs.py:137]: I think all dfs are not empty in our test cases. And this case is more complicated :)
This is the initial place that I tried to find what is causing the warning `CRS not set for some of the concatenation inputs.*`, and I still do not know how it is triggered :). The warning persisted when I set all dfs as pandas dataframe, so I guess a df with geom column (with crs) is still considered a gdf for some functions.

I simply silence the warning here, as this warning is only for information purposes, and I think we should not substantially change the code to fix it.

File: trackintel/preprocessing/staypoints.py (hunk 2 above)
Comment (hong2223) [reply trackintel/preprocessing/staypoints.py:276]: Yes! if a df is empty it will be considered as having no crs. setting a crs to an empty df will also cause an error. 

A more natural fix would be separating the cases.

File: trackintel/model/util.py (hunk 1 above)
Comment (hong2223) [reply trackintel/model/util.py:37]: You might be right. But I am not sure why a warning is raised :), see below (or above) comments.

**Review by hong2223 — COMMENTED** [review]
Thanks for the review! The places with comments are all "temporary" or "compromising" fixes. These warnings are all for information purposes, and I did not spend too much time figuring out their real causes.

File: trackintel/model/util.py (hunk 1 above)
Comment (NinaWie) [reply trackintel/model/util.py:37]: Seems like this was a bug in the past but is fixed now: https://github.com/geopandas/geopandas/issues/1648

File: trackintel/preprocessing/staypoints.py (hunk 2 above)
Comment (NinaWie) [reply trackintel/preprocessing/staypoints.py:276]: So you would add the warning for now, and then we work on the issue in another PR?

**Review by NinaWie — APPROVED** [review]
So, I had a closer look at this, but actually I can't reproduce any of the "CRS not set for some of the concatenation inputs" - warnings :D So it seems that it's again a version problem. I'm using geopandas 0.10.2, maybe they fixed the crs problems now.
I guess we can leave the catch_warnings() statements and add a comment that this is due to bugs in the old geopandas version? Otherwise it's fine for me to merge

File: trackintel/preprocessing/staypoints.py (hunk 2 above)
Comment (hong2223) [reply trackintel/preprocessing/staypoints.py:276]: Yes! a todo comment will be added

File: trackintel/model/util.py (hunk 1 above)
Comment (hong2223) [reply trackintel/model/util.py:37]: The warning persists in our test environment that uses geopandas 0.12.2

**Review by hong2223 — COMMENTED** [review]
This issue might be related, but the warning persists in our test environment that uses geopandas 0.12.2. I can also reproduce it on my location machine with the same version.

**Review by NinaWie — APPROVED** [review]
Looks good, thanks! Let's merge :)
